src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -425,27 +437,72 @@ generate_key_string(const std::string& argument_name, xrt_core::patcher::buf_typ
   return argument_name + buf_string;
 }
 
+
+// Basic Itanium ABI type decoding. demangle_arg_type() is referenced from ChatGPT response
 static std::string
-demangle(const std::string& mangled_name)
+demangle_arg_type(const std::string& s, size_t& idx)
```

Comment:
We don't need this function.  Leverage get_demangle_type and pass the char to get the type for.  Leave string (s) and index (idx) with caller.